### PR TITLE
doc: Literally include one-liner script in the cookbook

### LINKED ISSUE #2824: Literally include one-liner scripts in the cookbook
**Description of the desired feature**

In the [Map Projections](https://docs.generic-mapping-tools.org/dev/cookbook/map-projections.html) chapter of the cookbook, some maps are generated by one-liner scripts in the `doc/scripts` directory. These one-liner scripts use `-ps` to generate PS files for testing purposes. However, we want to show `-pdf` in our cookbook, thus we can't literally include them in the cookbook.

Since all scripts are converted to txt files by script2verbatim, I think we can update script2verbatim.c to allow it to replace `-ps` with `-pdf` during the conversion.

**Are you willing to help implement and maintain this feature?** Yes

### PR DESCRIPTION
**Pull request by seisman**
Rewrite `script2verbatim.c` to allow replacing `-ps` with `-pdf`,
and include one-liner scripts in the cookbook.

Closes #2824.

### CODE REVIEW

**Comment by seisman** [comment]
In `script2verbatim.c`, I use `gmt_strrep` to replace ` -ps ` with ` -pdf `. 

The build fails on Windows:
```
FAILED: src/script2verbatim.exe 
cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_exe --intdir=src\CMakeFiles\script2verbatim.dir --rc="C:\PROGRA~2\Windows Kits\10\bin\10.0.18362.0\x64\rc.exe" --mt="C:\PROGRA~2\Windows Kits\10\bin\10.0.18362.0\x64\mt.exe" --manifests  -- C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1424~1.283\bin\Hostx64\x64\link.exe /nologo src\CMakeFiles\script2verbatim.dir\script2verbatim.c.obj src\CMakeFiles\script2verbatim.dir\common_string.c.obj  /out:src\script2verbatim.exe /implib:src\script2verbatim.lib /pdb:src\script2verbatim.pdb /version:0.0  /machine:x64  /MANIFEST:NO setargv.obj /INCREMENTAL:NO /subsystem:console  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cmd.exe /C "cd /D D:\a\1\s\build\src && powershell -noprofile -executionpolicy Bypass -file C:/vcpkg/scripts/buildsystems/msbuild/applocal.ps1 -targetBinary D:/a/1/s/build/src/script2verbatim.exe -installedDir C:/vcpkg/installed/x64-windows/bin -OutVariable out""
LINK: command "C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1424~1.283\bin\Hostx64\x64\link.exe /nologo src\CMakeFiles\script2verbatim.dir\script2verbatim.c.obj src\CMakeFiles\script2verbatim.dir\common_string.c.obj /out:src\script2verbatim.exe /implib:src\script2verbatim.lib /pdb:src\script2verbatim.pdb /version:0.0 /machine:x64 /MANIFEST:NO setargv.obj /INCREMENTAL:NO /subsystem:console kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib" failed (exit code 1120) with the following output:
   Creating library src\script2verbatim.lib and object src\script2verbatim.exp
LINK : warning LNK4217: symbol 'gmt_strrep' defined in 'common_string.c.obj' is imported by 'script2verbatim.c.obj' in function 'main'
common_string.c.obj : error LNK2019: unresolved external symbol __imp_strndup referenced in function gmt_strdup_noquote
src\script2verbatim.exe : fatal error LNK1120: 1 unresolved externals
```
Maybe I should link to a Windows library?

**Comment by PaulWessel** [comment]
Hm, we have that one coded up in common_string.c so GMT gets it that way.

File: src/script2verbatim.c [hunk 1]
```diff
@@ -25,64 +25,165 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <assert.h>
```
Comment (seisman) [inline src/script2verbatim.c]:
```suggestion
```

**Comment by seisman** [comment]
I use `gmt_strrep` to replace `-ps` with `-pdf` in `script2verbatim.c`. `gmt_strrep` is defined in `common_string.c`, so I also linked this C file. 

As shown above, it failed on Windows, because one of the functions in common_string.c uses function `strndup`, which is natively provided by MSVC.

GMT has its own `strndup` in "gmt_notposix.c", but "gmt_notposix.h" needs "gmt_dev.h". It makes the dependencies complicated.

The easiest way is copying the entire `gmt_strrep` to `script2verbatim.c`, which works pretty well.

**Pull request merged**